### PR TITLE
Acm Store Collection Container (Issue #265)

### PR DESCRIPTION
**Pull request by annsudhart**
image

### CODE REVIEW

File: src/actions/storeActions.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    const data = await collectionsRes.json();
+
+    if (!data) throw new Error('Empty response from server');
+    if (data.error) throw new Error(data.error.message);
+    dispatch({
+        type: FETCH_COLLECTIONS,
+        payload: data.collections,
+    });
+} catch (error) {
+    notify('Unable to fetch future events!', error.message);
```
Comment (leestanley) [inline src/actions/storeActions.ts:39]: Should this say store here?

File: src/components/StoreCollection/style.less [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+.collection {
+
+    .collection-info {
+        margin: 15px 0px;
+    }
+
+    h2 {
+        font-size: @general-heading-2;
+        font-weight: bold;
```
Comment (leestanley) [inline src/components/StoreCollection/style.less:11]: Avoid using bold - just put the font-weight at 600 I believe

File: src/components/StoreItemCard/style.less [hunk 3]
```diff
@@ -0,0 +1,17 @@
+@import '../../styles/vars.less';
+
+.item-card{
+
+    border: 5px solid gray;
```
Comment (leestanley) [inline src/components/StoreItemCard/style.less:5]: Use `gray` your import from vars.less not the explicit name

File: src/actions/storeActions.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+            Authorization: `Bearer ${Storage.get('token')}`,
+        },  
+    });
+
+    const { status } = collectionsRes;
+    if (status === 401 || status === 403) {
+        dispatch(logoutUser());
```
Comment (leestanley) [inline src/actions/storeActions.ts:26]: hmm why dispatch a logout here versus a notify?

Comment (leestanley) [inline src/actions/storeActions.ts:26]: hmm why dispatch a logout here versus a notify?

Comment (annsudhart) [reply src/actions/storeActions.ts:26]: If they're not logged in, should we still show the store page and notify them that they're not logged in?

Comment (leestanley) [reply src/actions/storeActions.ts:26]: Oh interesting, do we not have a protected route from acmucsd.com/{store url} that already blocks people from viewing though?

Comment (StormFireFox1) [reply src/actions/storeActions.ts:26]: It is protected. As of now, it requires standard access, so no logged out user will see it anyway.

```jsx
            <Route
              exact
              path="/store"
              component={requireStandardAccess(requireAuth(StorePage)) as React.FC}
            />
```

See [index.tsx for routes](https://github.com/annsudhart/membership-portal-ui/blob/87b1b3e2cc6290e487f616a486a59f2bae544c4b/src/index.tsx#L87-L91).

File: src/actions/storeActions.ts (hunk 1 above)
Comment (StormFireFox1) [reply src/actions/storeActions.ts:39]: A more accurate error message would be welcome here. Maybe something like:

```suggestion
    notify('Unable to fetch store collections!', error.message);
```

File: src/components/StoreCollection/index.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            discountPercentage: number;
+            itemName: string;
+            lifetimeLimit: number;
+            monthlyLimit: number;
+            picture: string;
+            price: number;
+            uuid: string;
+        }
+    ]
+}
```
Comment (StormFireFox1) [inline src/components/StoreCollection/index.tsx:25]: We need to start thinking about that "types" library.

File: src/components/StorePage/index.tsx [hunk 6]
```diff
@@ -1,17 +1,16 @@
...
 
 const StorePage: React.FC = () => {
   return (
     <div className="store-page">
-      <h1>ACM Store</h1>
-      <div className="coming-soon">
-        <Icon className="temp-icon" component={SwagIcon} />
-        <p>Coming Soon!</p>
-      </div>
+      <h1>Diamond Outfitters</h1>
```
Comment (StormFireFox1) [inline src/components/StorePage/index.tsx:12]: I wonder if this is could be a dynamic property.

Comment (annsudhart) [reply src/components/StorePage/index.tsx:12]: Are you talking about displaying a different element depending on the store's status (if the store has nothing for sale/can't fetch collections/under maintenance)?

File: src/actions/storeActions.ts (hunk 4 above)
Comment (annsudhart) [reply src/actions/storeActions.ts:26]: I'll remove the code for the dispatch then if it's already taken care of

File: src/components/StorePage/index.tsx (hunk 6 above)
Comment (StormFireFox1) [reply src/components/StorePage/index.tsx:12]: I was more or less referring to the store name changing depending on store name; for instance, maybe we want the store to be called something else after a while. It turns out, however, that this is supposed to be replaced with the store header component anyway, so it doesn't matter.

**Pull request merged**